### PR TITLE
Add tests to bring coverage to 80%

### PR DESCRIPTION
**Pull request by maryfrances01**

### CODE REVIEW

**Comment by maryfrances01** [comment]
@Ajpantuso Just to note, when I ran pr_check, all tests passed, but when I did a git commit, I ran into this error:
ERRO Running error: 1 error occurred:
	* can't run linter goanalysis_metalinter: goanalysis_metalinter: buildir: package "netip" (isInitialPkg: false, needAnalyzeSource: true): in net/netip.AddrFromSlice: cannot convert Load <[]byte> t0 ([]byte) to [4]byte

Not sure if it is specific to my machine setup or does golangci-lint need to be updated?

**Comment by Ajpantuso** [comment]
> @Ajpantuso Just to note, when I ran pr_check, all tests passed, but when I did a git commit, I ran into this error: ERRO Running error: 1 error occurred: * can't run linter goanalysis_metalinter: goanalysis_metalinter: buildir: package "netip" (isInitialPkg: false, needAnalyzeSource: true): in net/netip.AddrFromSlice: cannot convert Load <[]byte> t0 ([]byte) to [4]byte
> 
> Not sure if it is specific to my machine setup or does golangci-lint need to be updated?

Typically running `pre-commit autoupdate` will fix that. It's usually because of the combination of golang version and golangci-lint version.

File: gocmd/gocmd_test.go [hunk 1]
```diff
@@ -19,3 +19,62 @@ func TestModule(t *testing.T) {
...
+		return tidyOptionFunc(func(c *TidyConfig) {
+			c.GoVersion = version
+		})
+	}
+
+	withWorkingDir := func(dir string) TidyOption {
+		return tidyOptionFunc(func(c *TidyConfig) {
+			c.WorkingDir = dir
+		})
+	}
```
Comment (Ajpantuso) [inline gocmd/gocmd_test.go:38]: It's probably better to just implement these as options [here](https://github.com/mt-sre/go-ci/blob/42a771b0ab5ed350136ba1db144cbe5351862c1d/gocmd/options.go). Then you can use them in tests or as a user of the library.

File: web/web_test.go [hunk 2]
```diff
@@ -33,4 +35,60 @@ func TestDownloadFile(t *testing.T) {
 	require.NoError(t, err)
 
 	assert.Equal(t, expectedContent, data)
+
+	tests := []struct {
+		name        string
+		url         string
+		out         string
+		expectError bool
```
Comment (Ajpantuso) [inline web/web_test.go:43]: nit: these fields are public to the struct so they should start with capital letters. Doesn't really matter in the context of tests, but it's a good habit to maintain that should make refactors easier in the future.

File: web/web_test.go [hunk 3]
```diff
@@ -33,4 +35,60 @@ func TestDownloadFile(t *testing.T) {
...
+			if tc.name == "server error" {
+				srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+					http.Error(w, "server error", http.StatusInternalServerError)
+				}))
+				defer srv.Close()
+				tc.url = srv.URL
+			}
+
+			err := web.DownloadFile(context.Background(), tc.url, out)
+			if tc.expectError {
```
[the remark this answers is not in the document]
Comment (maryfrances01) [reply web/web_test.go]: @Ajpantuso  OK - I think I did this correctly, but let me know if not!

File: web/web_test.go [hunk 4]
```diff
@@ -15,6 +15,20 @@ import (
...
+        t.Fatalf("expected no error but got: %v", err)
+    }
+}
+
+func Error(t *testing.T, err error) {
+    if err == nil {
+        t.Fatalf("expected an error but got none")
+    }
+}
+
```
Comment (Ajpantuso) [inline web/web_test.go]: You don't really need these helper funcs you can just use `require.NoError` and `require.Error` respectively.

File: web/web_test.go [hunk 5]
```diff
@@ -37,49 +51,49 @@ func TestDownloadFile(t *testing.T) {
...
 
 	tests := []struct {
-		name        string
-		url         string
-		out         string
-		expectError bool
+		Name         string
+		Url          string
+		Out          string
+		ExpectError  bool
```
Comment (Ajpantuso) [inline web/web_test.go]: This field can be removed now.

**Review by Ajpantuso — CHANGES_REQUESTED** [review]

File: web/web_test.go (hunk 4 above)
Comment (maryfrances01) [reply web/web_test.go]: @Ajpantuso OK - I see how this works now.

**Review by Ajpantuso — APPROVED** [review]
LGTM

**Pull request merged**